### PR TITLE
Don't retry one shot jobs during hive shutdown

### PR DESCRIPTION
**Pull request by giorio94**
Stop retrying to execute the function of a one shot job as soon as the job context is canceled, as the hive is already shutting down at that point. Indeed, the expectation is that the function exits immediately when the context expires, and it is most likely to return an error in that condition. Retrying would only delay the hive shutdown (also due to the additional backoff delay between consecutive attempts).

### CODE REVIEW

**Comment by giorio94** [comment]
Converting back to draft while investigating test failures.

**Comment by giorio94** [comment]
conformance-ginkgo hit https://github.com/cilium/cilium/issues/13071. Restarting the affected matrix entry.

File: pkg/hive/job/job.go [hunk 1]
```diff
@@ -277,15 +278,24 @@ func (jos *jobOneShot) start(ctx context.Context, wg *sync.WaitGroup, options op
 
 	stat := &spanstat.SpanStat{}
 
+	timer, cancel := inctimer.New()
+	cancel()
```
Comment (rgo3) [inline pkg/hive/job/job.go]:
```suggestion
	defer cancel()
```

**Review by rgo3 — CHANGES_REQUESTED** [review]

File: pkg/hive/job/job.go (hunk 1 above)
Comment (giorio94) [reply pkg/hive/job/job.go]: Good catch, thanks!

**Review by rgo3 — APPROVED** [review]

**Comment by rgo3** [comment]
Not really sure, deferring the question to @dylandreimerink or @joamaki :grimacing:

**Pull request merged**